OrbitQt/orbitmainwindow.cpp

```diff
@@ -282,6 +287,8 @@ OrbitMainWindow::OrbitMainWindow(QApplication* a_App,
     ui->actionIntrospection->setVisible(false);
   }
 
+  SetupGrpcChannel(std::move(grpc_server_address));
```

Comment:
What's the reason for the change? Conceptually, I feel that the MainWindow shouldn't need to own the grpc_channel, it should only own the UI. In an ideal world, Orbit would work headless without an instance of MainWindow (*dreaming of unicorns and rainbows*).

Reply:
The short answer is: It was simpler this way. 
With this change, GrpcChannel, ProcessManager and Process now live in TargetConfiguration, inside of OrbitMainWindow instead of App, but also only as long as the MainWindow exists. Before the MainWindow is destroyed, the Target is moved out of MainWindow, so it can be moved into ProfilingTargetDialog again and get reused. 
Some parts of Target are not relevant for App (and currently cant even be added because of dependencies between components, e.g. ServiceDeployManager). So if some parts would live in MainWindow and some in App, it would make the moving more complicated. 

I would argue that in a "headless" version, these things that form a Target still have to come from somewhere, if the user does not choose them by clicking in a UI, they come from cmdline arguments or whatever. They can live then in this layer and are getting put into App as its done here. I dont see this making this significantly more complicated in the future